pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -325,3 +311,37 @@ func k8sClusterConfig() (*rest.Config, error) {
 
 	return clusterConfig, nil
 }
+
+func NewK8sSync(
```

Comment:
Minor (personal preference) - I would like to see this on the top of the file so it is obvious that we have a New function to construct the structure.